rosbag2_cpp/test/rosbag2_cpp/test_reindexer.cpp

```diff
@@ -0,0 +1,129 @@
+// Copyright 2021 DCS Corporation, All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// DISTRIBUTION A. Approved for public release; distribution unlimited.
+// OPSEC #4584.
+//
+// Delivered to the U.S. Government with Unlimited Rights, as defined in DFARS
+// Part 252.227-7013 or 7014 (Feb 2014).
+//
+// This notice must appear in all copies of this file and its derivatives.
+
+#include <gmock/gmock.h>
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "rcpputils/asserts.hpp"
+#include "rcpputils/filesystem_helper.hpp"
+
+#include "rosbag2_cpp/readers/sequential_reader.hpp"
+#include "rosbag2_cpp/reindexer.hpp"
+
+#include "rosbag2_storage/bag_metadata.hpp"
+#include "rosbag2_storage/metadata_io.hpp"
+#include "rosbag2_storage/topic_metadata.hpp"
+
+#include "mock_converter.hpp"
+#include "mock_converter_factory.hpp"
+#include "mock_metadata_io.hpp"
+#include "mock_storage.hpp"
+#include "mock_storage_factory.hpp"
+
+using namespace testing;  // NOLINT
+
+class ReindexTestFixture : public Test
+{
+public:
+  ReindexTestFixture()
+  {
+    database_path = _SRC_REINDEX_DIR_PATH;
+    target_dir = database_path + "/target_metadata";
+  }
+
+  std::string database_path;
+  std::string target_dir;
+};
+
+TEST_F(ReindexTestFixture, test_multiple_files) {
+  auto bag_dir = database_path + "/multiple_files";
+  std::unique_ptr<rosbag2_cpp::reindexers::Reindexer> reindexer =
+    std::make_unique<rosbag2_cpp::reindexers::Reindexer>();
+
+  rosbag2_storage::StorageOptions so = rosbag2_storage::StorageOptions();
+  so.uri = bag_dir;
+  so.storage_id = "sqlite3";
+
+  reindexer->reindex(so);
+
+  // We should have a metadata file at this point
+  auto generated_file = rcpputils::fs::path(bag_dir + "/metadata.yaml");
+  EXPECT_TRUE(generated_file.exists());
+
+  // We made it here. Excellent. Truly excellent.
```

Comment:
I'm glad you're having fun, but let's tone down the comments across the board to "only if necessary to understand the code (and sometimes maybe the code needs clarifying rather than comments), and only as much information as needed" - the flavor commentary doesn't always translate well across all readers and distracts from the point at hand